src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/service/MigrationQueueService.kt

```diff
@@ -48,13 +56,32 @@ class MigrationQueueService(
 
   private fun Any.toJson() = objectMapper.writeValueAsString(this)
   fun purgeAllMessages() {
-    hmppsQueueService.purgeQueue(
-      PurgeQueueRequest(
-        queueName = migrationQueue.queueName,
-        sqsClient = migrationSqsClient,
-        queueUrl = migrationQueueUrl
+    // try purge first, since it is rate limited fall back to less efficient read/delete method
+    kotlin.runCatching {
+      hmppsQueueService.purgeQueue(
+        PurgeQueueRequest(
+          queueName = migrationQueue.queueName,
+          sqsClient = migrationSqsClient,
+          queueUrl = migrationQueueUrl
+        )
       )
-    )
+    }.onFailure {
+      deleteAllMessages()
+    }
+  }
+
+  private fun deleteAllMessages() {
+    kotlin.runCatching {
+      val messageCount = migrationSqsClient.countMessagesOnQueue(migrationQueueUrl)
+      repeat(messageCount) {
```

Comment:
is this count always going to be accurate?

Reply:
Probably not but this is called multiple times whe. Cancelling